dbt/models/default/default.vw_pin_value.sql

```diff
@@ -1,190 +1,472 @@
 -- duplicates.
-WITH stage_values AS (
+stage_values AS (
     SELECT
-        parid AS pin,
-        taxyr AS year,
+        asmt.parid AS pin,
+        asmt.taxyr AS year,
+        -- Pre-mailed values
+        ARBITRARY(
+            CASE
+                WHEN
+                    {{ pre_mailed_filters('asmt') }}
+                    THEN REGEXP_REPLACE(asmt.class, '[^[:alnum:]]', '')
+            END
+        ) AS pre_mailed_class,
+        ARBITRARY(
+            CASE
+                WHEN
+                    {{ pre_mailed_filters('asmt') }} AND asmt.taxyr < '2020'
+                    THEN asmt.ovrvalasm2
+                WHEN
+                    {{ pre_mailed_filters('asmt') }} AND asmt.taxyr >= '2020'
+                    THEN asmt.valasm2
+            END
+        ) AS pre_mailed_bldg,
+        ARBITRARY(
+            CASE
+                WHEN
+                    {{ pre_mailed_filters('asmt') }} AND asmt.taxyr < '2020'
+                    THEN asmt.ovrvalasm1
+                WHEN
+                    {{ pre_mailed_filters('asmt') }} AND asmt.taxyr >= '2020'
+                    THEN asmt.valasm1
+            END
+        ) AS pre_mailed_land,
+        ARBITRARY(
+            CASE
+                WHEN
+                    {{ pre_mailed_filters('asmt') }} AND asmt.taxyr < '2020'
+                    THEN asmt.ovrvalasm3
+                WHEN
+                    {{ pre_mailed_filters('asmt') }} AND asmt.taxyr >= '2020'
+                    THEN asmt.valasm3
+            END
+        ) AS pre_mailed_tot,
+        -- Pre-mailed market values
+        ARBITRARY(
+            CASE
+                WHEN
+                    {{ pre_mailed_filters('asmt') }} AND asmt.taxyr < '2020'
+                    THEN NULL
+                WHEN
+                    {{ pre_mailed_filters('asmt') }} AND asmt.taxyr >= '2020'
+                    THEN asmt.valapr2
+            END
+        ) AS pre_mailed_bldg_mv,
+        ARBITRARY(
+            CASE
+                WHEN
+                    {{ pre_mailed_filters('asmt') }} AND asmt.taxyr < '2020'
+                    THEN NULL
+                WHEN
+                    {{ pre_mailed_filters('asmt') }} AND asmt.taxyr >= '2020'
+                    THEN asmt.valapr1
+            END
+        ) AS pre_mailed_land_mv,
+        ARBITRARY(
+            CASE
+                WHEN
+                    {{ pre_mailed_filters('asmt') }} AND asmt.taxyr < '2020'
+                    THEN NULL
+                WHEN
+                    {{ pre_mailed_filters('asmt') }} AND asmt.taxyr >= '2020'
+                    THEN asmt.valapr3
+            END
+        ) AS pre_mailed_tot_mv,
         -- Mailed values
         ARBITRARY(
             CASE
                 WHEN
-                    procname = 'CCAOVALUE'
-                    THEN REGEXP_REPLACE(class, '[^[:alnum:]]', '')
+                    asmt.procname = 'CCAOVALUE'
+                    THEN REGEXP_REPLACE(asmt.class, '[^[:alnum:]]', '')
```

Comment:
It's annoying, but we have to adjust all of the non-provisional column definitions to prefix column names with aliases. This is due to the new join with the `stages` CTE we defined above. No logic should have changed for these non-provisional columns except for the addition of aliases.